sklearn/compose/tests/test_target.py

```diff
@@ -332,3 +332,22 @@ def test_transform_target_regressor_route_pipeline():
     pip.fit(X, y, **{'est__check_input': False})
 
     assert regr.transformer_.fit_counter == 1
+
+
+class DummyRegressorWithExtraPredictParams(DummyRegressor):
+    def predict(self, X, check_input=True):
+        # In the test below we make sure that the check input parameter is
+        # passed as false
+        assert not check_input
```

Comment:
This test will pass if `TransformedTargetRegressor` never calls `predict`. We need to add a custom attribute here to makes sure `predict` was called.

```suggestion
        self.predict_counter = getattr(self, "predict_counter", 0) + 1
        assert not check_input
```

and below in `test_transform_target_regressor_pass_extra_predict_parameters`

```py
regr.predict(X, check_input=False)
assert regr.regressor_.predict_counter == 1
```